authors/apps/follow/views.py

```diff
@@ -0,0 +1,101 @@
+                                 follower=self.request.user)
+                follow.save()
+                return Response({'success': 'Now following {}.'.format(
+                                following)}, status=status.HTTP_201_CREATED)
+        return Response({'error': 'Unable to create a following. '
+                        'This user does not exist. Please choose another '
+                         'user.'}, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request, user):
+        """Returns a list of followers for a given user."""
+        followers_list = Follows.objects.filter(followed_user=user)
+        serializer = FollowersSerializer(followers_list, many=True)
+        return Response({"followers": serializer.data},
+                        status=status.HTTP_200_OK)
+
+    def delete(self, request, user, follower):
+        """Removes a follower from a user's following. Checks if user
+        attempts to delete followers unrelated to them. It then confirms if the
+        given user actually follows the given follower."""
+        check_user = self.request.user.username
+        if check_user != user:
+            return Response({'error': 'Incorrect user logged in. '
+                            'Check username in the URL.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        if Follows.objects.filter(followed_user=follower).filter(
+                                  following_user=user).exists():
+            Follows.objects.filter(followed_user=follower).filter(
+                                  following_user=user).delete()
+            return Response({"success": '{} has been unfollowed.'.format(
+                            follower)}, status=status.HTTP_200_OK)
+        return Response({"error": 'You do not follow {}. Unfollow failed.'
+                        .format(follower)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ManageFollowings(APIView):
+    """Contains views related to a user's follows. This includes a method that
+    returns a list of all the followed users. Only authenticated users may
+    access the view."""
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, user):
+        """Returns a list of followed users for a given user."""
+        followings_list = Follows.objects.filter(following_user=user)
+        serializer = FollowingSerializer(followings_list, many=True)
+        return Response({"followed_users": serializer.data},
+                        status=status.HTTP_400_BAD_REQUEST)
+
+
+class UserStats(APIView):
+    """Contains views related to calculating statistics based on a user's
+    followers and follows. This includes a method that counts the number of
+    followers and follows of a given user."""
+    def get(self, request, user):
+        """Returns a count of a user's followers and follows."""
+        if User.objects.filter(username=user).exists():
+            amount_follows = Follows.objects.filter(
```

Comment:
Shouldn't this view also update the number of followings/followers in the Profiles model since we included it there too?

Reply:
I have addressed the issue in my newest PR.